### PR TITLE
fix some unhandled promises and refactor some code

### PR DESCRIPTION
**Pull request by dkanada**

### CODE REVIEW

**Review by MrTimscampi — APPROVED** [review]
LGTM.

File: src/components/dialogHelper/dialogHelper.js [hunk 1]
```diff
@@ -106,15 +106,6 @@ import '../../assets/css/scrollstyles.css';
                     tryRemoveElement(dlg);
                 }
             }
-
-            //resolve();
-            // if we just called history.back(), then use a timeout to allow the history events to fire first
-            setTimeout(() => {
```
Comment (thornbill) [inline src/components/dialogHelper/dialogHelper.js:112]: Why was this removed?

File: src/components/alert.js [hunk 2]
```diff
@@ -33,14 +33,7 @@ import globalize from '../scripts/globalize';
...
             options.buttons = items;
-
-            return dialog.show(options).then(function (result) {
-                if (result === 'ok') {
-                    return Promise.resolve();
-                }
-
-                return Promise.reject();
-            });
+            return dialog.show(options);
```
Comment (thornbill) [inline src/components/alert.js:36]: Doing a quick search, it looks like there are other places that would be handling this promise rejection. How did you confirm this is not needed?

**Review by thornbill — CHANGES_REQUESTED** [review]

File: src/components/dialogHelper/dialogHelper.js (hunk 1 above)
Comment (dkanada) [reply src/components/dialogHelper/dialogHelper.js:112]: I was seeing tons of unhandled exceptions for dialogs and traced it back to this weird section of code, and removing it had no ill effects. History is already broken on dialogs in stable and I didn't want to look into it too much, so I figured it was fine.

File: src/components/alert.js (hunk 2 above)
Comment (dkanada) [reply src/components/alert.js:36]: I actually found the opposite, can you point to an example? It looked to me like there were two or three methods with the same name, and this one wasn't using the promise rejection anywhere. It was actually another source of unhandled exceptions whenever the dialog was closed without explicitly clicking the OK button. The only use I saw was the one to supress an unhandled exception.

**Review by thornbill — APPROVED** [review]
